services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/controller/GSController.java

```diff
@@ -0,0 +1,173 @@
+   *
+   * @param gs Encapsulation of NetworkConfiguration message
+   * @return RestOperationResult in String
+   * @throws Exception Various exceptions that may occur during the create process
+   * @link https://github.com/haboy52581/alcor/blob/master/docs/modules/ROOT
+   *     /pages/infra_services/data_plane_manager.adoc
+   */
+  @PostMapping({"/subnet/", "v4/subnet/"})
+  @ResponseStatus(HttpStatus.CREATED)
+  public List<InternalDPMResult> createSubnet(@RequestBody NetworkConfiguration gs)
+      throws Exception {
+    gs.setOpType(Common.OperationType.CREATE);
+    gs.setRsType(Common.ResourceType.SUBNET);
+    return service(gs);
+  }
+
+  /**
+   * Accept north bound calls then transfer to ACA calls in southbound
+   *
+   * @param gs Encapsulation of NetworkConfiguration message
+   * @return RestOperationResult in String
+   * @throws Exception Various exceptions that may occur during the update process
+   * @link https://github.com/haboy52581/alcor/blob/master/docs/modules/ROOT
+   *     /pages/infra_services/data_plane_manager.adoc
+   */
+  @PutMapping({"/subnet/", "v4/subnet/"})
+  public List<InternalDPMResult> updateSubnet(@RequestBody NetworkConfiguration gs)
+      throws Exception {
+    gs.setOpType(Common.OperationType.UPDATE);
+    gs.setRsType(Common.ResourceType.SUBNET);
+    return service(gs);
+  }
+
+  /**
+   * Accept north bound calls then transfer to ACA calls in southbound
+   *
+   * @param gs Encapsulation of NetworkConfiguration message
+   * @return RestOperationResult in String
+   * @throws Exception Various exceptions that may occur during the delete process
+   * @link https://github.com/haboy52581/alcor/blob/master/docs/modules/ROOT
+   *     /pages/infra_services/data_plane_manager.adoc
+   */
+  @DeleteMapping({"/subnet/", "v4/subnet/"})
+  public List<InternalDPMResult> deleteSubnet(@RequestBody NetworkConfiguration gs)
+      throws Exception {
+    gs.setOpType(Common.OperationType.DELETE);
+    gs.setRsType(Common.ResourceType.SUBNET);
+    return service(gs);
+  }
+
+  private List<InternalDPMResult> service(NetworkConfiguration gs) {
+    // TODO: Create a verification framework for all resources
+    // leave isFast as true since SB GSinfo does not have fastpath attr
+    try {
+      return goalStateManager
+          .talkToACA(
+              goalStateManager.transformNorthToSouth(gs),
+              true,
+              Integer.parseInt(config.getPort()),
+              Boolean.valueOf(config.getOvs()))
+          .stream()
+          .flatMap(Collection::stream)
+          .map(
+              f -> {
+                return new InternalDPMResult(
+                    f.getResourceId(),
+                    f.getResourceType().toString(),
+                    f.getOperationStatus().toString(),
+                    f.getStateElapseTime());
+              })
+          .collect(Collectors.toList());
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
Please use our alcor logger for any logging need.

Reply:
fixed in af1b619